protocols/ospfv3/packet/lsa.go

```diff
@@ -0,0 +1,526 @@
+package packet
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/bio-routing/bio-rd/util/decode"
+	"github.com/bio-routing/tflow2/convert"
+	"github.com/pkg/errors"
+)
+
+type LSA struct {
+	Age               uint16
+	Type              LSType
+	ID                ID
+	AdvertisingRouter ID
+	SequenceNumber    uint32
+	Checksum          uint16
+	Length            uint16
+	Body              Serializable
+}
+
+const LSAHeaderLength = 20
+
+func (x *LSA) Serialize(buf *bytes.Buffer, headerOnly bool) {
```

Comment:
"A flag argument is a kind of function argument that tells the function to carry out a different operation depending on its value." https://martinfowler.com/bliki/FlagArgument.html
Let's make it 2 functions.